### ISSUE TITLE
Pagemaster (SGB) - no sound effects

### ISSUE THREAD

**Issue by Sun-in-jang** [visualboyadvance-m/visualboyadvance-m#405]
VisualBoyAdvance-M 2.1.1

No sound in Pagemaster in sound test and in-game, when you picking items (like keys), there only music all the time. Almost the same in Animaniacs (SGB) - sounds working, but no music.

**Comment by denisfa** [comment]
@rkitover Both these games seems to be working correctly on latest git build. I verified sounds/music from youtube's walkthroughs, and they match with what I am hearing when playing. I tested on Linux x64 and only the beginning of both, though.

@Sun-in-jang could you provide some more info so we can debug this better? What platform are you playing? If it is windows, could you download and try our newer builds?

- [Windows 32 bit](https://github.com/visualboyadvance-m/visualboyadvance-m/files/3044710/visualboyadvance-m-Win-32bit.zip)
- [Windows 64 bit](https://github.com/visualboyadvance-m/visualboyadvance-m/files/3044712/visualboyadvance-m-Win-64bit.zip)

**Comment by rkitover** [comment]
also please tell us if you're using a bios file, I should add this to the template

**Comment by Sun-in-jang** [comment]
Platform - Windows 64 bit, tested newer build - the same result. Bios file is not using (i'm unable to find it).

**Comment by denisfa** [comment]
@Sun-in-jang Thank you for the feedback!  I just want to confirm something:

1. Is this the game [Animaniacs](https://www.youtube.com/watch?v=a-i5cQNQz7Y)?
2. Is this the other [Pagemaster](https://www.youtube.com/watch?v=N4mUd8h30yk)?

If that is the case, could you please test one more time, but running the configuration file attached.
[vbam.zip](https://github.com/visualboyadvance-m/visualboyadvance-m/files/3051002/vbam.zip)

In case you are not aware where the configuration file is, type `%applocaldata%` on the menu explorer, find the folder `visualboyadvance-m` and replace the file `vbam` (move to a different location for backup) inside with the one from the attached zip.

Also, BIOS files are mostly illegal unless you have permission to have one, so remove your zip if that is not the case.

**Comment by Sun-in-jang** [comment]
Yes these games is correct. Tested with configuration file, music and sounds fine

**Comment by denisfa** [comment]
@Sun-in-jang Thank you very much for the information and testing! We will see what we can do about this.

**Comment by retro-wertz** [comment]
since this is an sgb-compatible game, if this will be using the snes audio api, then that is not supported, or never will be... (it requires emulating snes audio)

**Comment by rkitover** [comment]
notes from irc:

> \<kode54\> yup, sounds like they need Higan
> \<kode54\> or current bsnes, which also has the SGB module
> \<mudlord\> or snes_faust

**Comment by Sun-in-jang** [comment]
Looks like bsnes is best one for me. Higan supported sgb, but i don't know how to emulate these games with it (even no video on youtube). Want to ask you what is "snes_faust"

**Comment by retro-wertz** [comment]
Want to ask you what is "snes_faust"
mednafen

**Comment by rkitover** [comment]
we can also try to steal the SGB code from bsnes

**Comment by retro-wertz** [comment]
gl...

**Comment by Sun-in-jang** [comment]
Will be waiting any news on about this topic (can be able to test newer builds)

**Comment by Squall-Leonhart** [comment]
VBA is not a super nintendo emulator, mimicking the graphics output is one thing, the audio is an entirely different ballgame.
